common/src/field_conversion.rs

```diff
@@ -2,11 +2,16 @@ use ark_bn254::Fr as ArkFr;
 use ark_ff::{fields::PrimeField as ArkPrimeField, BigInteger};
 
 use ff::PrimeField as GenericPrimeField;
-use spartan2::provider::bn256_grumpkin::bn256::Base as Spartan2Fr;
-
-pub fn ark_to_spartan(ark: ArkFr) -> Spartan2Fr {
-    let bigint = ark.into_bigint();
-    Spartan2Fr::from_raw(bigint.0)
+use spartan2::provider::bn256_grumpkin::bn256::Scalar as Spartan2Fr;
+
+pub fn ark_to_spartan<ArkF: ArkPrimeField>(ark: ArkF) -> Spartan2Fr {
```

Comment:
Does this work for the generic version? `ark_to_ff`

If so let's convert that one as well.

Reply:
Yup. Works with that conversion function.